DiscordChatExporter.Core/Exporting/Filtering/HasMessageFilter.cs

```diff
@@ -0,0 +1,25 @@
+﻿using DiscordChatExporter.Core.Discord.Data;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace DiscordChatExporter.Core.Exporting.Filtering
+{
+    public class HasMessageFilter : MessageFilter
+    {
+        private readonly string _value;
+
+        public HasMessageFilter(string value) => _value = value;
+
+        public override bool Filter(Message message) =>
+            _value switch
+            {
+                "link" => Regex.IsMatch(message.Content, "https?://\\S*[^\\.,:;\"\'\\s]", DefaultRegexOptions),
+                "embed" => message.Embeds.Any(),
+                "file" => message.Attachments.Any(),
+                "video" => message.Attachments.Any(file => file.IsVideo),
+                "image" => message.Attachments.Any(file => file.IsImage),
+                "sound" => message.Attachments.Any(file => file.IsAudio),
+                _ => false
```

Comment:
I forgot if we discussed this, but maybe it's better to throw here

```suggestion
                _ => throw new InvalidOperationException($"Invalid value provided for the 'has' message filter: '{_value}'");
```

Reply:
I think we discussed whether `from:`/`mentions:` should throw with invalid input, but now that we accept usernames all input is valid for them. I agree that this one should throw; I'll make the change.